mlir/lib/Conversion/MeshToMPI/MeshToMPI.cpp

```diff
@@ -832,9 +924,10 @@ struct ConvertMeshToMPIPass
 
     patterns.add<ConvertUpdateHaloOp, ConvertNeighborsLinearIndicesOp,
                  ConvertProcessMultiIndexOp, ConvertGetShardingOp,
-                 ConvertShardingOp, ConvertShardShapeOp>(typeConverter, ctxt);
-    // ConvertProcessLinearIndexOp accepts an optional worldRank
-    patterns.add<ConvertProcessLinearIndexOp>(typeConverter, ctxt, worldRank);
+                 ConvertShardingOp, ConvertShardShapeOp, ConvertAllReduceOp,
+                 ConvertProcessLinearIndexOp>(typeConverter, ctxt);
+    SymbolTableCollection symbolTableCollection;
+    mlir::mesh::populateFoldingPatterns(patterns, symbolTableCollection);
```

Comment:
Reusing non-conversion patterns in a conversion is generally incorrect. The conversion patterns need to access potentially changed values through the adaptor, which you don't have access to from the folding patterns.

Are these foldings really required for the conversion to succeed? If not, this should be kept as a separate pass that runs either before or after.

Reply:
The conversion requires the folding patterns. I had my own conversion patterns and later replaced them with the folders to avoid duplication of functionality.

I could refactor: outline the actual implementation and use that in the folder and a separate conversion pattern. However, this looks overly complicated, I would hope there is a better way to do this.